package-lock.json

```diff
@@ -0,0 +1,5675 @@
+{
```

Comment:
I personally like not having locks in libraries. But I’m up for changing my mind.

Reply:
It’s particularly interesting to have them if everything is bundled (so practicly no dependencies, only devdependencies). But bundling seems weird for a nodejs library and lockfiles will not be honored by npm while installing this lib within another project. I agree that the lock maybe omitted, unless we opt not to have dependencies but only devdependencies.

Reply:
@wooorm What's your rationale behind not including a lock file for libs?

Reply:
I think it just makes commit history dirty. This comment would be helpful.
https://github.com/sindresorhus/ama/issues/479#issuecomment-310661514

---

And, we could omit it easily by adding `.npmrc`

```
package-lock=false
```

Reply:
Yeah, I think Sindre sums it up nicely. And I don’t agree with the Yarn post by James.

I ignore `package-lock.json` through an [`.npmrc`](https://github.com/remarkjs/remark-rehype/blob/master/.npmrc), and `yarn.lock` from [`.gitignore`](https://github.com/remarkjs/remark-rehype/blob/07469598e3a2c794f51e0741b8213066b3c702c9/.gitignore#L8).

Reply:
My counterargument would be as follows — we can’t reliably know what versions of the dependencies our user will install unless we have exact versions in our `package.json`. The possibility of the code in this package running with different versions of its dependencies is inevitable. Therefore, irrespective of the existence of a lockfile, bugs caused by this are inevitable. Also, @sindresorhus’ comment about publishing a buggy version due to variance in dependency versions doesn’t go away by removing the lock file. On the contrary, having a lockfile helps figure out faster if a version drift caused the bug. 

Having a lockfile at least makes the dependency tree known during development. This comes with benefits such as faster installs, `npm ci`, potentially faster debugging as we can rule out dependency update related bugs faster, etc. 

As I see it, having a lockfile doesn’t impact our users but gives us certain benefits during dev. Whereas not having it rids us of these benefits while not having any impact on our users. Therefore, I’d vote to commit the file.

Reply:
The argument was about not having the same view as your user’s. Having a lockfile is excellent for developers but gives false confidence once your ci installs the exact same dependencies as in every dev’s machine. The ci should reflect the real world, what would happen if somebody installs your package and tries it. If one of your dependencies introduces a breaking change, devs and ci won’t see it but your users will.

You can always use a lockfile without committing it and you’ll get all benefits of a lockfile locally. As for having exact dependency versions, it won’t solve the issue: one of you dependencies may not use exact version for one of its dependencies.
The only way to guarantee that users will always get the same package without conflict with the external world would to either bundle the lib or to not use dependencies. Not having the lockfile inside the repo will make us detect issues as they happen.

@wooorm @zeusdeux @Rokt33r would having at least one frequent ci job/sanity check ignore the lockfile be a good compromise ?

Reply:
I’ll withdraw my vote in support of committing the lockfile as there’s momentum against it and it feels like I’m fueling bikeshedding. 
I’ll remove em in a commit that follows. If we find we see value in having it in the future, we can add it back in. 
Sound good?

Reply:
Sweet, we can revisit it later if need be!